src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -93,6 +94,33 @@ void java_bytecode_convert_classt::convert(const classt &c)
   }
 
   java_class_typet class_type;
+  if(c.signature.has_value() && c.signature.value()[0]=='<')
+  {
+    java_generics_class_typet generic_class_type;
+#ifdef DEBUG
+    std::cout << "INFO: found generic class signature "
+              << c.signature.value()
+              << " in parsed class "
+              << c.name << "\n";
+#endif
+    try
+    {
+      const std::vector<typet> &generic_types=java_generic_type_from_string(
+        id2string(c.name),
+        c.signature.value());
+      for(const typet &t : generic_types)
+      {
+        generic_class_type.generic_types()
+          .push_back(to_java_generic_parameter(t));
+      }
+      class_type=generic_class_type;
+    }
+    catch(unsupported_java_class_siganture_exceptiont)
+    {
+      // Do nothing: we don't support parsing for example double bounded or
```

Comment:
maybe here we should fall back to using the descriptor, and emit a warning, in order to degrade gracefully

Reply:
My understanding is the `descriptor` won't have any generic type information so isn't applicable here (We are only trying to fill out generic information for the class)

Reply:
Adding a warning is a good idea however.